### PR TITLE
Fix typo in is_valid_uuid regex

### LINKED ISSUE #16: Typo in regex in `is_valid_uuid()`
### Bug Description

I think we are having an issue in the method `is_valid_uuid()` of `juju_topology.py`. Seems the regex has a typo, and [unit test are failing](https://github.com/canonical/prometheus-k8s-operator/runs/6889022732?check_suite_focus=true#step:5:567).

```python
    def is_valid_uuid(self, uuid):
        """Validates the supplied UUID against the Juju Model UUID pattern."""
        regex = re.compile(
            "^[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}$"
        )
        return bool(regex.match(uuid))
```

With [this  :point_up_2:  regex the UUID `f2c1b2a6-e006-11eb-ba80-0242ac130004` is not matching](https://regex101.com/r/5cf52U/1), but with [this one UUIDs matches](https://regex101.com/r/hAfpvF/1).

### To Reproduce

Just run unit test in any charm that uses `juju_topology.py`

### Environment

edge

### Relevant log output

```shell
_____ TestActiveStatus.test_unit_is_blocked_if_reload_configuration_fails ______
Traceback (most recent call last):
  File "/home/runner/work/prometheus-k8s-operator/prometheus-k8s-operator/tests/unit/test_charm_status.py", line 78, in test_unit_is_blocked_if_reload_configuration_fails
    self.harness.begin_with_initial_hooks()
  File "/home/runner/work/prometheus-k8s-operator/prometheus-k8s-operator/.tox/unit/lib/python3.8/site-packages/ops/testing.py", line 225, in begin_with_initial_hooks
    self.begin()
  File "/home/runner/work/prometheus-k8s-operator/prometheus-k8s-operator/.tox/unit/lib/python3.8/site-packages/ops/testing.py", line 188, in begin
    self._charm = TestCharm(self._framework)
  File "/home/runner/work/prometheus-k8s-operator/prometheus-k8s-operator/src/charm.py", line 62, in __init__
    self._scraping = MetricsEndpointProvider(
  File "/home/runner/work/prometheus-k8s-operator/prometheus-k8s-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 1410, in __init__
    self.topology = JujuTopology.from_charm(charm)
  File "/home/runner/work/prometheus-k8s-operator/prometheus-k8s-operator/lib/charms/observability_libs/v0/juju_topology.py", line 144, in from_charm
    return cls(
  File "/home/runner/work/prometheus-k8s-operator/prometheus-k8s-operator/lib/charms/observability_libs/v0/juju_topology.py", line 120, in __init__
    raise InvalidUUIDError(model_uuid)
charms.observability_libs.v0.juju_topology.InvalidUUIDError: 'f2c1b2a6-e006-11eb-ba80-0242ac130004' is not a valid UUID.
```

### Additional context

With this change, UUIDs like `f2c1b2a6-e006-11eb-ba80-0242ac130004` are validated.

```diff
@@ -128,7 +128,7 @@ class JujuTopology:
     def is_valid_uuid(self, uuid):
         """Validates the supplied UUID against the Juju Model UUID pattern."""
         regex = re.compile(
-            "^[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}$"
+            "^[a-f0-9]{8}-?[a-f0-9]{4}-?[a-f0-9]{4}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}$"
         )
         return bool(regex.match(uuid))

```

### PR DESCRIPTION
**Pull request by Abuelodelanada**
## Issue

This PR fixes #16 

## Solution

Fix a typo in the regex that validates UUID

## Context

The change is simple:

```diff
@@ -128,7 +128,7 @@ class JujuTopology:
     def is_valid_uuid(self, uuid):
         """Validates the supplied UUID against the Juju Model UUID pattern."""
         regex = re.compile(
-            "^[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}$"
+            "^[a-f0-9]{8}-?[a-f0-9]{4}-?[a-f0-9]{4}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}$"
         )
         return bool(regex.match(uuid))
```

## Testing Instructions

Run unit test. We added a new one: `test_invalid_uuid`

## Release Notes

Fix typo in the regular expression that validates UUID in the method `is_valid_uuid`

### CODE REVIEW

**Review by rbarry82 — DISMISSED** [review]

File: tests/test_juju_topology.py
Comment (sed-i) [inline tests/test_juju_topology.py:103]: Consider adding standard uuids to test against.

```python
        for invalid_uuid in (
            str(uuid.uuid1()), 
            str(uuid.uuid4()), 
            "f2c1b2a6-006-11eb-ba80-0242ac130004"
        ):
            with self.subTest(invalid_uuid=invalid_uuid):
                topology_invalid_uuid = OrderedDict(...)
                # ...
```

**Review by dstathis — DISMISSED** [review]

**Review by dstathis — APPROVED** [review]

**Review by sed-i — APPROVED** [review]
Maybe better to merge this after the ops fix is merged.
But then reqs.txt would need to have github again.

**Pull request merged**
